Gutenberg/Gutenberg.podspec

```diff
@@ -0,0 +1,73 @@
+#
+# This is used to fetch Gutenberg builds that come from commits instead of tags.
+# Builds from tags are "official" and stable, so we distribute them via GitHub we'll eventually publish them on CocoaPods.
+# The artifacts for builds from commits are instead only stored on an Automattic's server.
+Pod::Spec.new do |s|
+  raise 'Could not find Gutenberg version configuration' if GUTENBERG_CONFIG.nil?
+
+  gutenberg_version = GUTENBERG_CONFIG[:commit]
+
+  raise "Trying to fetch Gutenberg XCFramework from Automattic's distribution server with invalid version '#{GUTENBERG_CONFIG}'" if gutenberg_version.nil?
+
+  xcframework_storage_url = 'https://d2twmm2nzpx3bg.cloudfront.net'
+
+  xcframework_archive_url = "#{xcframework_storage_url}/Gutenberg-#{gutenberg_version}.tar.gz"
+
+  require 'net/http'
+
+  raise "Could not find file at URL #{xcframework_archive_url}" unless Net::HTTP.get_response(URI(xcframework_archive_url)).code == '200'
+
+  s.name = 'Gutenberg'
+  s.version = '1.0.0' # The value here is irrelevant, but required
+  s.summary = 'A spec to help integrating the Gutenberg XCFramework'
+  s.homepage = 'https://apps.wordpress.com'
+  s.license = { type: 'GPL', file: '../LICENSE' }
+  s.authors = 'Automattic'
+
+  s.ios.deployment_target = '13.0' # TODO: Read from common source
+  s.swift_version = '5.0' # TODO: read from common source
+
+  s.requires_arc = true # TODO: Can this be omitted?
+
+  # Tell CocoaPods where to download the XCFramework(s) archive with `source` and what to use from its decompressed contents with `vendored_frameworks`.
+  #
+  # Unfortunately, CocoaPods currently (1.12.1) does not work when it comes to local specs with http sources.
+  #
+  # See https://github.com/CocoaPods/CocoaPods/issues/10288#issuecomment-1517711223
+  # s.source = { http: xcframework_archive_url }
+  archive_name = "Gutenberg-#{gutenberg_version}.tar.gz"
+  # Always use relative paths, otherwise the checksums in the lockfile will change from machine to machine
+  relative_extracted_archive_directory = Pathname.new("#{GUTENBERG_ARCHIVE_DIRECTORY}").relative_path_from(__dir__).to_s
+  relative_download_directory = Pathname.new(GUTENBERG_DOWNLOADS_DIRECTORY).relative_path_from(__dir__).to_s
+  relative_download_path = File.join(relative_download_directory, archive_name)
+
+  s.source = { http: "file://#{relative_download_path}" }
+
+  s.vendored_frameworks = [
+    'Aztec.xcframework',
+    'Gutenberg.xcframework',
+    'React.xcframework',
+    'RNTAztecView.xcframework',
+    'yoga.xcframework'
+  ].map do |f|
+    # This needs to be a relative path to the local extraction location and account for the archive folder structure.
+    File.join(relative_extracted_archive_directory, 'Frameworks', f)
+  end
+
+  # Print the message here because the prepare_command output is not forwarded by CocoaPods
+  puts "Will skip downloading Gutenberg archive because it already exists at #{relative_download_path}" if File.exist? relative_download_path
```

Comment:
Here's how it looks in action

<img width="1268" alt="Screenshot 2023-06-02 at 2 06 26 pm" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/5f0ee988-83be-4889-be8d-c664c9b446fb">


Admittedly, that's not very visible. Plus, using the relative path makes it confusing to know where the file actually is. I plan to iterate on this, but I'm saving it for later because going back and forth on this setup often requires paying the download cost of a fresh gutenberg-mobile repo with submodules.

Once both commit and tag install are supported, we won't have to download that repo anymore and we'll be able to iterate faster on this.